terraform/moe_bucket.tf

```
@@ -8,3 +8,70 @@ resource "aws_s3_bucket" "moe_bucket" {
     prevent_destroy = true
   }
 }
+
+# IAM policy document that that allows read permissions on the MOE bucket.
+data "aws_iam_policy_document" "moe_bucket_read_doc" {
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "s3:ListBucket",
+    ]
+
+    resources = [
+      aws_s3_bucket.moe_bucket.arn,
+    ]
+  }
+
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "s3:GetObject",
+    ]
+
+    resources = [
+      "${aws_s3_bucket.moe_bucket.arn}/*",
+    ]
```

I think we can safely consolidate this to:
```suggestion
    actions = [
      "s3:GetObject",
      "s3:ListBucket",
    ]

    resources = [
      aws_s3_bucket.moe_bucket.arn,
      "${aws_s3_bucket.moe_bucket.arn}/*",
    ]
```

---

I'm fine either way.

---

So this was based (I think) on the [AWS example IAM policy](https://docs.aws.amazon.com/AmazonS3/latest/userguide/example-policies-s3.html#iam-policy-ex0) and given some of the [more advanced examples](https://docs.aws.amazon.com/AmazonS3/latest/userguide/example-policies-s3.html#iam-policy-ex1) I'm wary of combining these permissions and having possibly unexpected interactions. Thoughts?

---

I'm fine keeping it as is.  If it ain't broke....

---

Just for documentation I did test both IAM policy changes and expected functionality worked.